src/hoprd/hoprd_ingress.rs

```diff
@@ -20,20 +17,26 @@ use serde_json::json;
 use std::{collections::BTreeMap, sync::Arc};
 use tracing::{error, info};
 
-use crate::{hoprd::hoprd_ingress, model::Error as HoprError};
 use crate::{constants, context_data::ContextData, operator_config::IngressConfig, utils};
+use crate::{hoprd::hoprd_ingress, model::Error as HoprError};
 
 use super::hoprd_service::ServiceTypeEnum;
 
 /// Creates a new Ingress for accessing the hoprd node,
-pub async fn create_ingress(context_data: Arc<ContextData>, service_type: &ServiceTypeEnum, service_name: &str, namespace: &str, ingress_config: &IngressConfig, owner_references: Option<Vec<OwnerReference>>) -> Result<i32, Error> {
-    let labels: Option<BTreeMap<String, String>> = Some(utils::common_lables(
-        context_data.config.instance.name.to_owned(),
-        Some(service_name.to_owned()),
-        None,
-    ));
+pub async fn create_ingress(
+    context_data: Arc<ContextData>,
+    service_type: &ServiceTypeEnum,
+    service_name: &str,
+    namespace: &str,
+    session_port_allocation: &i32,
+    ingress_config: &IngressConfig,
+    owner_references: Option<Vec<OwnerReference>>,
+) -> Result<i32, Error> {
+    let labels: Option<BTreeMap<String, String>> = Some(utils::common_lables(context_data.config.instance.name.to_owned(), Some(service_name.to_owned()), None));
     let p2p_port = if service_type.eq(&ServiceTypeEnum::ClusterIP) {
-        hoprd_ingress::open_port(context_data.client.clone(), &namespace, &service_name, &context_data.config.ingress).await.unwrap()
+        hoprd_ingress::open_port(context_data.client.clone(), &namespace, &service_name, session_port_allocation, &context_data.config.ingress)
+            .await
+            .unwrap()
```

Comment:
Does not work because of 
`?` couldn't convert the error to `kube::Error`
the question mark operation (`?`) implicitly performs a conversion on the error value using the `From` trait
the following other types implement trait `FromResidual<R>`:
  <Result<T, F> as FromResidual<Yeet<E>>>
  <Result<T, F> as FromResidual<Result<Infallible, E>>>
required for `Result<i32, kube::Error>` to implement `FromResidual<Result<Infallible, model::Error>>`rustc[Click for full compiler diagnostic](rust-analyzer-diagnostics-view:/diagnostic message [0]?0#file:///Users/ausias/Documents/github/hoprd-operator/src/hoprd/hoprd_ingress.rs)